src/lib/arch/aarch64/Instruction_execute.cc

```diff
@@ -907,19 +908,19 @@ void Instruction::execute() {
           out[i] += ((n[i] >> j) & 1);
         }
       }
-      results[0] = out;
+      results[0] = {out, 256};
       break;
     }
     case Opcode::AArch64_CPYi32: {  // dup vd, vn.s[index]
       const uint32_t* vec = operands[0].getAsVector<uint32_t>();
-      uint32_t out[4] = {vec[metadata.operands[1].vector_index], 0, 0, 0};
-      results[0] = out;
+      uint32_t out[1] = {vec[metadata.operands[1].vector_index]};
+      results[0] = {out, 256};
```

Comment:
could reducing the size of the output array not lead to random data being left in the vector instead of being set to `0`?

Reply:
I'm not sure what you mean by random data being left in the vector, could you please clarify?

The 0s are present to pad the array up to 16 bytes. And as explained below, the `RegisterValue` will be zero-extended to 256 bytes at the end of the function. I'm not sure why it is 16 bytes, perhaps because the registers for FP/NEON instruction       are 16 bytes.

The arrays no longer need to padded with the updated constructor as they are automatically zero extended to 256 bytes.

Reply:
I didn't realise that it would be zero-extended at the end of the function.... thanks again for clarification